packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -11,188 +11,238 @@
+     */
+    featureTestSelector: LocalFeatureTestSelector,
+  ) =>
+  <LocalQueryMethod extends QueryMethod = "get">(
+    props: {
+      /**
+       * Refers to Testing Library's canvas. This is usually `screen`, but Storybook uses `within(canvas)`.
+       */
+      element: HTMLElement;
+    } & InnerQuerySelectorProps<LocalFeatureTestSelector, LocalQueryMethod>,
+  ) => {
+    const { element: containerElement, queryMethod } = props;
+    const localQueryMethod = queryMethod || ("get" as const);
+    const querySelectorOptions = "options" in props ? props.options : undefined;
+    const role = "role" in props ? (props.role as AriaRole) : undefined;
+
+    // This `let` is difficult to make into a `const`. It makes the code unreadable.
+    let capturedElement: HTMLElement | null = null;
+
+    if ("selector" in featureTestSelector && querySelectorOptions) {
+      const sharedProps = {
+        element: containerElement,
+        queryMethod: localQueryMethod,
+        queryOptions: Object.fromEntries(
+          Object.entries(featureTestSelector.selector.options).map(
+            ([testSelectorsKey, testingLibraryKey]) => [
+              testingLibraryKey,
+              querySelectorOptions[testSelectorsKey],
+            ],
+          ),
+        ) as Record<
+          LocalFeatureTestSelector extends TestSelector
+            ? LocalFeatureTestSelector["selector"]["options"][keyof LocalFeatureTestSelector["selector"]["options"]]
+            : string,
+          string | RegExp
+        >,
+      };
+
+      capturedElement =
+        featureTestSelector.selector.method === "ByRole"
+          ? getByRoleQuerySelector({
+              ...sharedProps,
+              role:
+                Array.isArray(featureTestSelector.selector.role) || role
+                  ? role || ""
+                  : featureTestSelector.selector.role,
+            })
+          : getByTextQuerySelector({
+              ...sharedProps,
+              selectionMethod: featureTestSelector.selector.method,
+              text: featureTestSelector.selector.text,
+            });
+    } else if (
+      "isControlledElement" in featureTestSelector &&
+      featureTestSelector.isControlledElement
+    ) {
+      try {
+        capturedElement = getControlledElement({ element: containerElement });
+      } catch (error) {
+        if (queryMethod === "query") {
+          capturedElement = null;
+        }
+
+        throw error;
+      }
+    }
+
+    if (!capturedElement) {
+      throw new ElementError(
+        "No child HTML element available",
+        containerElement,
+      );
+    }
+
+    if (!("accessibleText" in featureTestSelector)) {
+      throw new Error("Missing `accessibleText` in `FeatureTestSelector`");
+    }
+
+    const getAccessibleText = <
+      LabelName extends LocalFeatureTestSelector extends AccessibleLabelSelector
+        ? keyof LocalFeatureTestSelector["accessibleText"]
+        : keyof AccessibleLabelSelector,
```

Comment:
Is this conditional correct? keyof AccessibleLabelSelector would only ever be `accessibleText` right? Where as the branch above it would be whatever string is in that record. Is `accessibleText` an expected labelName here?

Reply:
I changed it to `never` for the situation you don't have `accessibleText` defined on your object.